### DISCUSSION TITLE
Change invite mode via DB

### DISCUSSION THREAD

**Discussion by leahoswald** [mastodon/mastodon#19808] [Q&A]
We want to change the invite mode in the DB to implement a script to stop invite creation by normal users after a specific number of new users per day. Therefore we want to change it from user to admin and back. It looks like this is done by the min_invite_role value in the settings table. But changing this value doesn't change anything (eg. what the admin settings shows). Is there any other setting in the database we missed to change? (we use mastodon v3.5.3)

**Comment by rixx** [comment]
To add: we are aware that this setting will change and will use `user_roles` once mastodon v4 is out, but we'd like to change this now. Any hints would be appreciated!

**Reply by rixx** [reply]
Tried out some more things, and it's looking like changing things in the `settings` table is never mirrored in the frontend, but changing things in the frontend is always mirrored in the `settings` table. Is there some sort of caching going on? If so, how do we trigger a cache clear in the application?

I tried this:

```sql
update
  settings
set
  value = replace(value, 'user', 'admin'),
  updated_at = timezone('utc', now())
where
  var = 'min_invite_role';
```

The update looks just like what the website does, as far as I can tell.

*Edit:* Ah, yes, found it – the settings are also set and gotten from redis, from a key structured `cache:rails_settings_cached/<id>/min_invite_role`.
